services:
  pg:
    image: postgres
    container_name: ai_pg
    hostname: pg.dev
    environment:
      POSTGRES_DB: bot
      POSTGRES_USER: user
      POSTGRES_PASSWORD: password
    ports:
      - "5432:5432"
    volumes:
      - pgdata:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready"]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - app-network
  pgadmin:
    image: dpage/pgadmin4
    container_name: ai_pgadmin
    environment:
      PGADMIN_DEFAULT_EMAIL: admin@admin.org
      PGADMIN_DEFAULT_PASSWORD: admin
      PGADMIN_CONFIG_SERVER_MODE: 'False'
    ports:
      - "5050:80"
    networks:
      - app-network
  app:
    build:
      context: .
      dockerfile: dockerfile
    container_name: ai_bot
    environment:
      - PYTHONUNBUFFERED=1
    env_file: .env
    depends_on:
      pg:
        condition: service_healthy
    networks:
      - app-network

volumes:
  pgdata:

networks:
  app-network:
